bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/FindReplaceOverlay.java

```diff
@@ -0,0 +1,870 @@
+				() -> findReplaceLogic.performReplaceAll(getFindString(), getReplaceString()));
+	}
+
+	private void performSelectAll() {
+		BusyIndicator.showWhile(getShell() != null ? getShell().getDisplay() : Display.getCurrent(),
+				() -> findReplaceLogic.performSelectAll(getFindString()));
+	}
+
+	private KeyListener shortcuts = KeyListener.keyPressedAdapter(e -> {
+		e.doit = false;
+		if ((e.stateMask & SWT.CTRL) != 0 && (e.keyCode == 'F' || e.keyCode == 'f')) {
+			close();
+		} else if ((e.stateMask & SWT.CTRL) != 0 && (e.keyCode == 'R' || e.keyCode == 'r')) {
+			if (findReplaceLogic.getTarget().isEditable()) {
+				toggleReplace();
+			}
+		} else if ((e.stateMask & SWT.CTRL) != 0 && (e.keyCode == 'W' || e.keyCode == 'w')) {
+			toggleToolItem(wholeWordSearchButton);
+		} else if ((e.stateMask & SWT.CTRL) != 0 && (e.keyCode == 'P' || e.keyCode == 'p')) {
+			toggleToolItem(regexSearchButton);
+		} else if ((e.stateMask & SWT.CTRL) != 0 && (e.keyCode == 'A' || e.keyCode == 'a')) {
+			toggleToolItem(searchInSelectionButton);
+		} else if ((e.stateMask & SWT.CTRL) != 0 && (e.keyCode == 'C' || e.keyCode == 'c')) {
+			toggleToolItem(caseSensitiveSearchButton);
+		} else if (e.keyCode == SWT.CR || e.keyCode == SWT.KEYPAD_CR) {
+			performEnterAction(e);
+		} else if (e.keyCode == SWT.ESC) {
+			close();
+		} else {
+			e.doit = true;
+		}
+	});
+
+	private void performEnterAction(KeyEvent e) {
+		boolean isShiftPressed = (e.stateMask & SWT.SHIFT) != 0;
+		boolean isCtrlPressed = (e.stateMask & SWT.CTRL) != 0;
+		if (okayToUse(replaceBar) && replaceBar.isFocusControl()) {
+			if (isCtrlPressed) {
+				performReplaceAllOnEnter();
+			} else {
+				performReplaceOnEnter();
+			}
+		} else {
+			if (isCtrlPressed) {
+				performSearchAltOnEnter();
+			} else {
+				performSearchOnEnter(isShiftPressed);
+			}
+		}
+	}
+
+	private void performReplaceAllOnEnter() {
+		performReplaceAll();
+		evaluateFindReplaceStatus();
+	}
+
+	private void performReplaceOnEnter() {
+		performSingleReplace();
+		evaluateFindReplaceStatus();
+	}
+
+	private void performSearchAltOnEnter() {
+		performSelectAll();
+		evaluateFindReplaceStatus();
+	}
+
+	private void performSearchOnEnter(boolean isShiftPressed) {
+		performSearch(!isShiftPressed);
+		evaluateFindReplaceStatus();
+	}
+
+	private void toggleToolItem(ToolItem toolItem) {
+		toolItem.setSelection(!toolItem.getSelection());
+		toolItem.notifyListeners(SWT.Selection, null);
+	}
+
+	private ControlListener shellMovementListener = new ControlListener() {
+		@Override
+		public void controlMoved(ControlEvent e) {
+			positionToPart();
+		}
+
+		@Override
+		public void controlResized(ControlEvent e) {
+			positionToPart();
+		}
+	};
+
+	private FocusListener overlayFocusListener = FocusListener.focusLostAdapter(e -> {
+			findReplaceLogic.activate(SearchOptions.GLOBAL);
+			searchInSelectionButton.setSelection(false);
+	});
+
+	private PaintListener widgetMovementListener = __ -> {
+			positionToPart();
+	};
```

Comment:
Just as a pointer for you (not a request for change), @Wittmaxi: You can write single line lambdas even more compact without an expression block.
```suggestion
	private PaintListener widgetMovementListener = __ -> positionToPart();
```

Reply:
Changing this is no big deal, I have addressed this here: https://github.com/eclipse-platform/eclipse.platform.ui/commit/808bea19300dbec2fda76f0f42584149dd2c044c

Thank you for all the suggestions :)